### PR TITLE
Adding iree_task_executor_create_from_flags.

### PR DESCRIPTION
**Pull request by benvanik**
Progress on #4298: cleaning up HAL driver registration.

### CODE REVIEW

**Comment by benvanik** [comment]
(I'll fix the mako flag usage after #5613 lands, which I suspect will happen before this)

**Comment by hanhanW** [comment]
> (I'll fix the mako flag usage after #5613 lands, which I suspect will happen before this)

Thanks! It passed on BuildKite so I think it's ready to land. I will review this PR soon.

File: iree/task/api.c [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        IREE_STATUS_INVALID_ARGUMENT,
+        "one of --task_topology_group_count or --task_topology_mode must be "
+        "specified and be a valid value; have --task_topology_mode=%s.",
+        FLAG_task_topology_mode);
+  }
+
+  if (iree_status_is_ok(status)) {
+    status = iree_task_executor_create(scheduling_mode, &topology,
+                                       host_allocator, out_executor);
+  }
```
Comment (hanhanW) [inline iree/task/api.c:119]: should we return earlier if it failed?

**Review by hanhanW — APPROVED** [review]

File: iree/task/api.c (hunk 1 above)
Comment (benvanik) [reply iree/task/api.c:119]: it needs to fall through to clean up the topology/end the trace zone regardless of whether it fails, so this lets it do that and then return the status (which may be ok or failed)

**Pull request merged**